Add `platform prerequisites` for ambient, and simplify `getting started` ambient guide (#14709)
istio/istio.io: content/en/docs/ops/ambient/install/platform-prerequisites/index.md

---
title: Platform-Specific Prerequisites
description: Platform-Specific Prerequisites for Installing Istio Ambient Mode.
weight: 4
owner: istio/wg-environments-maintainers
test: no
---

This document covers any platform or environment specific prerequisites for installing Istio in ambient mode.

## CNI

### Cilium

1. Cilium currently defaults to proactively deleting other CNI plugins and their config, and must be configured with

> `cni.exclusive = false` to properly support chaining. See https://docs.cilium.io/en/stable/helm-reference/ for more details.

Reviewer: ```suggestion
`cni.exclusive = false` to properly support chaining. See [the Cilium documentation](https://docs.cilium.io/en/stable/helm-reference/) for more details.
```

1. Due to how Cilium manages node identity and internally allow-lists node-level health probes to pods,

> applying default-DENY `NetworkPolicy` in a Cilium CNI install with Istio ambient mesh will cause `kubelet` health probes (which are by-default exempted from NetworkPolicy enforcement by Cilium) to be blocked.

Reviewer: ```suggestion
applying default-DENY `NetworkPolicy` in a Cilium CNI install underlying Istio in ambient mode, will cause `kubelet` health probes (which are by-default exempted from NetworkPolicy enforcement by Cilium) to be blocked.
```

---
title: Platform-Specific Prerequisites
description: Platform-Specific Prerequisites for Installing Istio Ambient Mode.
weight: 4
owner: istio/wg-environments-maintainers
test: no
---

This document covers any platform or environment specific prerequisites for installing Istio in ambient mode.

## Platform

### Minikube

> 1. If you are using [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) with the [docker driver](https://minikube.sigs.k8s.io/docs/drivers/docker/), you may need to append `--set cni.cniNetnsDir="/var/run/docker/netns"` to the `helm install` command so that the Istio `cni-agent`node agent can correctly manage and capture pods on the node.

Reviewer 2: Split long lines, and add a space between cni-agent and node.

```suggestion
1. If you are using [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) with the [docker driver](https://minikube.sigs.k8s.io/docs/drivers/docker/),
you may need to append `--set cni.cniNetnsDir="/var/run/docker/netns"` to the `helm install` command so that the Istio `cni-agent` node agent can correctly manage
and capture pods on the node.
```

Author: derp

---
title: Platform-Specific Prerequisites

> description: Platform-Specific Prerequisites for Installing Istio Ambient Mode.

Reviewer: ```suggestion
description: Platform-specific prerequisites for installing Istio in ambient mode.
```

Don't like the duplication but not blocking on it for now

weight: 4
owner: istio/wg-environments-maintainers
test: no
---

This document covers any platform or environment specific prerequisites for installing Istio in ambient mode.

## Platform

### Minikube

> 1. If you are using [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) with the [docker driver](https://minikube.sigs.k8s.io/docs/drivers/docker/),
> you may need to append `--set cni.cniNetnsDir="/var/run/docker/netns"` to the `helm install` command so that the Istio `cni-agent` node agent can correctly manage
> and capture pods on the node.

Reviewer: ```suggestion
1. If you are using [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) with the [Docker driver](https://minikube.sigs.k8s.io/docs/drivers/docker/),
you may need to append `--set cni.cniNetnsDir="/var/run/docker/netns"` to the `helm install` command so that the`istio-cni` node agent can correctly manage
and capture pods on the node.
```

Author: I am not 100% if this is required on macOS. It definitely is on Linux. Let me reboot into macOS and see.

Author: Changed `you may` to `you must` - it's required in all contexts I know of.

## CNI

### Cilium

1. Cilium currently defaults to proactively deleting other CNI plugins and their config, and must be configured with
`cni.exclusive = false` to properly support chaining. See [the Cilium documentation](https://docs.cilium.io/en/stable/helm-reference/) for more details.

1. Due to how Cilium manages node identity and internally allow-lists node-level health probes to pods,
applying default-DENY `NetworkPolicy` in a Cilium CNI install underlying Istio in ambient mode, will cause `kubelet` health probes (which are by-default exempted from NetworkPolicy enforcement by Cilium) to be blocked.

    This can be resolved by applying the following `CiliumClusterWideNetworkPolicy`:

    {{< text syntax=yaml snip_id=none >}}
    apiVersion: "cilium.io/v2"
    kind: CiliumClusterwideNetworkPolicy
    metadata:
      name: "allow-ambient-hostprobes"
    spec:
      description: "Allows SNAT-ed kubelet health check probes into ambient pods"
      endpointSelector: {}
      ingress:
      - fromCIDR:
        - "169.254.7.127/32"
    {{< /text >}}

>     Please see the [#49277](https://github.com/istio/istio/issues/49277) and [CiliumClusterWideNetworkPolicy](https://docs.cilium.io/en/stable/network/kubernetes/policy/#ciliumclusterwidenetworkpolicy) for more details.

Reviewer: ```suggestion
    Please see [issue #49277](https://github.com/istio/istio/issues/49277) and [CiliumClusterWideNetworkPolicy](https://docs.cilium.io/en/stable/network/kubernetes/policy/#ciliumclusterwidenetworkpolicy) for more details.
```
